File: internal/doc_engine/grounding/langmatrix/testdata/python/sample.py
"""Grounding fixture for the depth matrix."""
import asyncio
import os


# Service settings loaded from the environment.
class Config:
    """Service settings loaded from the environment."""

    def __init__(self):
        self.addr = os.environ.get("SAMPLE_ADDR", "localhost")
        self.port = int(os.getenv("SAMPLE_PORT", "8080"))


# Format a greeting for name; raises on empty input.
def format_greeting(name):
    """Format a greeting for name."""
    if not name:
        raise ValueError("greet: empty name")
    return "hello " + name


# Greet one name; keeps a call edge from greet_all.
async def greet_one(name):
    """Greet one name."""
    return format_greeting(name)


# Greet every name concurrently with asyncio.
async def greet_all(names):
    """Greet every name."""
    return await asyncio.gather(*[greet_one(n) for n in names])


# Test-looking function exercising format_greeting.
def test_format_greeting():
    """Exercise format_greeting."""
    assert format_greeting("ada") == "hello ada"

File: internal/doc_engine/grounding/langmatrix/testdata/ruby/sample.rb
# Grounding fixture for the depth matrix.
require "net/http"

# Service settings loaded from the environment.
class Config
  attr_reader :addr, :port
  def initialize
    @addr = ENV.fetch("SAMPLE_ADDR", "localhost")
    @port = ENV.fetch("SAMPLE_PORT", "8080")
  end
end

# Format a greeting; raises on empty input.
def format_greeting(name)
  raise ArgumentError, "greet: empty name" if name.nil? || name.empty?
  "hello #{name}"
end

# Greet one name; keeps a call edge from greet_all.
def greet_one(name)
  format_greeting(name)
end

# Greet every name on background threads.
def greet_all(names)
  names.map { |n| Thread.new { greet_one(n) }.value }
end

# Test-looking function exercising format_greeting.
def test_format_greeting
  raise "bad greet" unless format_greeting("ada") == "hello ada"
end
